components/scheduler/src/Scheduler.svelte

```diff
@@ -230,120 +246,34 @@
-    overflow: hidden;
-    display: grid;
-    font-family: Arial, Helvetica, sans-serif;
-    position: relative;
-    z-index: 1;
-
-    .booker {
-      height: calc(100% - 2rem);
-      overflow: auto;
-      background: rgba(0, 0, 0, 0.03);
-      padding: 1rem;
-
-      h2 {
-        font-size: 1.3rem;
-        font-weight: bold;
-        margin-bottom: 0.5rem;
-      }
-
-      & > p {
-        opacity: 0.8;
-        font-size: 0.9rem;
-        line-height: 1.3rem;
-        margin-bottom: 1rem;
-      }
-
-      .timeslots {
-        display: grid;
-        grid-auto-flow: row;
-        gap: 1rem;
-        margin-bottom: 1rem;
 
-        li {
-          background: rgba(255, 255, 255, 0.8);
-          list-style-type: none;
-          margin: 0;
-          padding: 1rem;
-          border: 1px solid #ebebeb;
-          display: grid;
-          gap: 0.5rem;
-          grid-auto-flow: row;
-
-          h3 {
-            font-size: 0.8rem;
-            opacity: 0.8;
-          }
-          .time {
-            color: var(--blue);
-          }
-          .date {
-            font-size: 0.8rem;
-          }
-
-          footer {
-            display: grid;
-            grid-auto-flow: row;
-            gap: 0.5rem;
-            padding: 1rem;
-            margin: 1rem -1rem -1rem;
-            font-size: 0.8rem;
-            background: #eeeef5;
-            strong {
-              display: block;
-            }
-
-            .cadences,
-            .expiry {
-              display: flex;
-              flex-wrap: wrap;
-              gap: 0.5rem;
-
-              label {
-                padding: 0.25rem 0.5rem;
-                transition: 0.1s;
-                background-color: #fff;
-                align-items: center;
-                display: flex;
-                gap: 0.5rem;
-                cursor: pointer;
-                &.checked {
-                  background-color: var(--blue);
-                  color: white;
-                }
-                input {
-                  padding-left: 5px;
-                  &.after {
-                    width: 15%;
-                  }
-                }
+  // #region custom fields
+  let customFieldResponses: Record<string, any> = {}; // ideally "any" would be "string | boolean", but text inputs cast to many types
+  $: if (
+    !custom_fields.find((field) =>
+      customFieldResponses.hasOwnProperty(field.title),
+    )
+  ) {
+    customFieldResponses = custom_fields.reduce((responses, field) => {
+      console.log("reca");
+      return { ...responses, [field.title]: "" };
+    }, {});
+  }
 
-                span {
-                  text-transform: capitalize;
-                }
-              }
-            }
-          }
-        }
-      }
+  $: requiredFieldsFilled = custom_fields
+    .filter((field) => field.required)
+    .every(
+      (field) =>
+        customFieldResponses[field.title] !== undefined &&
+        customFieldResponses[field.title] !== "",
+    );
```

Comment:
Boolean reactive value that depends on your required custom fields being filled.